textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/nlpextrator/NlpExtractor.java

```diff
@@ -42,21 +38,47 @@
     private List<Attribute> searchInAttributes;
     private ITuple sourceTuple;
     private Schema returnSchema;
+    private NlpConstants NlpConstant = null;
+    private String flag = null;
 
 
-    public static final String NE_NUMBER = "Number";
-    public static final String NE_LOCATION = "Location";
-    public static final String NE_PERSON = "Person";
-    public static final String NE_ORGANIZATION = "Organization";
-    public static final String NE_MONEY = "Money";
-    public static final String NE_PERCENT = "Percent";
-    public static final String NE_DATE = "Date";
-    public static final String NE_TIME = "Time";
+    /**
+     * Named Entity Constants: NE, Number, Location, Person, Organization, Money, Percent, Date, Time.
+     * Part Of Speech Constants: Noun, Verb, Adjective, Adverb
+     */
+    public enum NlpConstants {
+        NE, Number, Location, Person, Organization, Money, Percent, Date, Time, Noun, Verb, Adjective, Adverb;
+
+        private static boolean isPOSConstant(NlpConstants constant) {
+            if (constant.equals(NlpConstants.Adjective) || constant.equals(NlpConstants.Adverb) || constant.equals(NlpConstants.Noun) || constant.equals(NlpConstants.Verb)) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
 
+    ;
 
-    public NlpExtractor(IOperator operator, List<Attribute> searchInAttributes) {
+
+    /**
+     * @param operator
+     * @param searchInAttributes
+     * @param nlpConstant
+     * @throws DataFlowException
+     * @about The constructor of the NlpExtractor. Allow users to pass a list of attributes and a NlpConstant.
+     * The operator will only search within the attributes and return the same token that recognized as the same input
+     * NlpConstant. IF the input constant is NlpConstants.NE, return all tokens that recognized as NamedEntity Constants.
```

Comment:
"that recognized" -> "that are recognized".

Do you think it's better to rename `NlpConstants.NE` to `NlpConstants.ALL`?